examples/matching-engine/README.md

```diff
@@ -0,0 +1,143 @@
+<!-- cargo-rdme start -->
+
+# Matching Engine Example Application
+
+This sample application demonstrates a matching engine, showcasing the DeFi capabilities
+on the Linera protocol
+
+The matching engine trades between two tokens `token_0` & `token_1`, we can take the help
+of the `fungible` application example to create two token applications
+
+An order can be of two types:
+
+- Bid: For buying token 1 and paying in token 0, these are ordered in from the highest
+  bid (most preferable) to the smallest bid.
+- Ask: For selling token 1, to be paid in token 0, these are ordered from the smallest
+  (most preferable) to the highest bid.
+
+An `OrderId` is used to uniquely identify an order and enables following functionality
+
+- Modify: Allows to modify the order
+- Cancel: Cancelling an order
+
+When inserting an order it goes through the following steps
+
+- Transfer of tokens from the `fungible` application to the `matching engine` application through a cross application call so that it can be paid to the counterparty
```

Comment:
```suggestion
- Transfer of tokens from the `fungible` application to the `matching engine` application through a cross-application
call so that it can be paid to the counterparty
```